bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```diff
@@ -233,21 +230,15 @@ private LogContext configureLogContext() throws IOException {
     }
 
     public void run() throws Exception {
-        try {
-            server = buildServer(startServerArgs);
-        } catch (RuntimeException ex) {
-            cleanup();
-            throw ex;
-        }
-
         Runtime.getRuntime().addShutdownHook(new ShutdownHook());
-        server.start();
-    }
-
-    private void cleanup() {
-        log.deletingHome(environment.getJBossHome());
-        deleteDir(environment.getJBossHome());
+        server = buildServer(startServerArgs);
 
+        if (Files.notExists(pidFile)) {
+            Files.write(pidFile, Collections.singleton(Long.toString(org.wildfly.common.os.Process.getProcessId())), StandardCharsets.UTF_8, StandardOpenOption.CREATE_NEW);
+        } else {
+            log.pidFileAlreadyExists(pidFile, environment.getJBossHome());
```

Comment:
Maybe instead of just logging this should throw an exception. What do you think @jfdenise?

Reply:
I agree, that seems an invalid state. We should start from an empty directory.